### PR TITLE
INT-1911: Enable roundtripper logs

### PR DESCRIPTION
**Pull request by sairam-vudatha**
Tested consuming this change in shipping service. Seeing round tripper logs now.
Screenshot 2023-04-11 at 2 22 51 PM

### CODE REVIEW

File: api/request.go [hunk 1]
```diff
@@ -59,7 +62,7 @@ func (a API) makeRequestAndUnmarshalResponse(url string, request *models.Envelop
 		//   --> we DO NOT log an error, it is not an error from the logging perspective
 		//   --> this is still considered an error from the code-level perspective,
 		//       so we still return the error
-		if false == strings.Contains(err.Error(), "This tracking number cannot be found") {
+		if !strings.Contains(err.Error(), "This tracking number cannot be found") {
```
Comment (devin-haslam) [inline api/request.go:62]: 🤣

File: api/request.go [hunk 2]
```diff
@@ -6,22 +6,25 @@ import (
...
+	onlyOnce   sync.Once
+	logger     *log.Logger
+	httpClient *http.Client
 )
 
-func init() {
-	logger = logrus.WithFields(logrus.Fields{
-		"app": "fedex",
+func (a API) makeRequestAndUnmarshalResponse(url string, request *models.Envelope, response models.Response) error {
+	onlyOnce.Do(func() {
```
Comment (devin-haslam) [inline api/request.go]: Does it make more sense to add the httpClient to the API struct? Personally I hate the onlyOnce function, but maybe other engineers feel differently

Comment (sairam-vudatha) [reply api/request.go]: I agree that onlyOnce is not neat. But on the consuming side [we are building API struct from shippingMethod.creds json](https://github.com/happyreturns/shipping/blob/master/carriers/fedex/client_builder.go#L39). [We also have config file where we are building API struct from](https://github.com/happyreturns/shipping/blob/master/carriers/fedex/config.yaml#L1). So I went with onlyOnce to not change that part.

**Review by devin-haslam — APPROVED** [review]
look great!

**Pull request merged**